app/controllers/api/v1/products_controller.rb

```diff
@@ -29,8 +30,7 @@ def update
   end
 
   def create
-    category = Category.find(params.require(:category_id))
-    response = Product.insert(product_params, category)
+    response = Product.insert(product_params, @category)
```

Comment:
Why passing `@category` to a Product's class method?, this create a strong external dependency, there is another way to do this using `@category`.